src/main/java/seedu/address/logic/parser/ImportCommandParser.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import seedu.address.logic.commands.ImportCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ImportCommand object
+ */
+public class ImportCommandParser implements Parser<ImportCommand> {
+
+    private static final String MESSAGE_INVALID_COMMAND_FORMAT = "Invalid command format!\n";
+    private static final String MESSAGE_INVALID_PATH = "Invalid path!\n";
+
+    /**
+     * Parses the given import path as {@code String} and translates it into a Path object
+     * and returns an ImportCommand object for execution.
+     * @throws ParseException
+     */
+    public ImportCommand parse(String args) throws ParseException {
+        if (args.isBlank() || args.trim().isEmpty()) {
+            throw new ParseException(
+                String.format("%s\n%s", MESSAGE_INVALID_COMMAND_FORMAT, ImportCommand.MESSAGE_USAGE));
+        }
+        requireNonNull(args);
```

Comment:
Perhaps this should be before the if statement? If not a nullpointerexception could occur!

Reply:
based on how the parse works, null pointer seems impossible I've attempted to cause it with "import" and null inputs but to no avail. Thus I removed the catch block as it was unnecessary